File: INTRODUCCION A POO/EJERCICIO3/Producto.py
class Producto:
    def __init__(self, nombre, precio, stock):
        self.nombre = nombre
        self.precio = precio
        self.stock = stock
        
    def vender(self, cantidad):
        if cantidad <= self.stock:
            self.stock = self.stock - cantidad
            print("Se vendieron ",cantidad, self.nombre, "Quedan: ",self.stock, self.nombre)
        else:
            print("No hay suficiente producto")
    
    def reabastecer(self, cantidad):
        self.stock += cantidad
        print("se reabastecieron ", cantidad, self.nombre, "quedan: ",self.stock, self.nombre)
    
o1 = Producto("galletas", 2, 40)
print("El stock es: ",o1.stock)
o1.vender(5)
o1.reabastecer(8)

File: INTRODUCCION A POO/EJERCICIO3/Producto.java
package EJERCICIO3;

public class Producto {
    public String nombre;
    public double precio;
    public int stock;

    public Producto(String nombre, double precio, int stock){
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
    }

    public void vender(int cantidad){
        if (cantidad <= this.stock) {
            this.stock = this.stock - cantidad;
            System.out.println("se vendieron "+cantidad +" " +this.nombre+". Quedan: "+this.stock + " "+this.nombre);
            
        } else {
            System.out.println("No hay sufieciente producto");
        }
    }
    public void reabastecer(int cantidad){
        this.stock = this.stock + cantidad;
        System.out.println("se reabastecieron "+ cantidad +" "+ this.nombre +". Quedan: "+ this.stock + " "+this.nombre);
    }
}
